### PR TITLE
feat: add impression events

### PR DESCRIPTION
**Pull request by ivarconr**
## About the changes
the purpose of impression data is to make it easy to integrate with other analytics tools and internal data lakes ([read full write-up](https://www.getunleash.io/blog/impression-data-send-unleash-statistics-anywhere))

## Discussion points
Do we need a way to disable it for performance reasons? Need to validate if emitting events have any cost if nobody subscribes!

### CODE REVIEW

**Comment by ivarconr** [comment]
[Node SDK: Impression Data support](https://trello.com/c/OKet69dD/1276-node-sdk-impression-data-support)

**Review by sighphyre — APPROVED** [review]
Looks good to me overall, just a small language suggestion!

File: src/events.ts [hunk 1]
```diff
@@ -10,4 +12,18 @@ export enum UnleashEvents {
   CountVariant = 'countVariant',
   Sent = 'sent',
   Registered = 'registered',
+  Impression = 'impression'
+}
+
+export interface ImpressionEvent {
+  eventType: "isEnabled" | "getVariant";
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/events.ts:19]: I have no clue why this was allowed. Should have been single quotes. Fixing.

File: src/events.ts [hunk 2]
```diff
@@ -10,4 +12,18 @@ export enum UnleashEvents {
...
+  context: Context;
+  enabled: boolean;
+  featureName: string;
+  variant?: string;
+}
+
+// Wrapper to provide type checking.
+export function createImpressionEvent(evt: ImpressionEvent) {
```
[the remark this answers is not in the document]
Comment (ivarconr) [reply src/events.ts]: Added return type as well, even though it is implicit.

**Pull request merged**